src/parameterizations/vertical/MOM_geothermal.F90

```diff
@@ -136,6 +144,10 @@ subroutine geothermal(h, tv, dt, ea, eb, G, GV, CS, halo)
 !$OMP                                  wt_in_place,dTemp,dRcv,h_transfer,heating, &
 !$OMP                                  I_h)
 
+! Save temperature and thickness before any changes are made (for diagnostic)
+temp_old = tv%T
+h_old = h
```

Comment:
Just noticed this, loops need to be explicit in MOM6 so implicit assigments are generally not allowed.

But I will follow up on this in the next comment.

Reply:
OK thanks.

Reply:
I thought that it was 'allowed' in the MOM6 style guidelines if it was a simple assignment, but that the dimensions should be made clear?
`
temp_old(:,:,:) = tv%T(:,:,:)
h_old(:,:,:) = h(:,:,:)
`

Reply:
I think the [style guide](https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#array-syntax) only permits array syntax for initialization, since halo transfers must be explicit.

Reply:
Actually in this case you may be right @ashao, since it's just a copy.

Reply:
Ah yes fair enough. Don't let me lead you astray.

Reply:
In any case it's a good point of clarification for the style guide. Are array copies a 'Thou mayest' or 'Thou shalt not'?